File: dice_test.py
# dice_test.py
import os
import time
import random
from pathlib import Path
from playwright.sync_api import sync_playwright

def run_dice_fairness_tests():
    with sync_playwright() as p:
        browser = p.chromium.launch() # Launch a headless browser
        page = browser.new_page()

        # Navigate to the local HTML file
        html_file_path = Path(__file__).parent / 'index.html'
        page.goto(f"file://{html_file_path}")

        # Selectors for the elements we need to interact with
        die1_input = page.locator('#die1')
        die2_input = page.locator('#die2')
        roll_button = page.locator('#rollButton')
        reset_button = page.locator('#resetButton')
        fairness_score_display = page.locator('#fairnessScoreDisplay')

        # Handle the confirmation dialog for the reset button
        def handle_dialog(dialog):
            if dialog.type == 'confirm' and 'לאפס את כל נתוני הזריקות' in dialog.message:
                dialog.accept()
            else:
                dialog.dismiss()
        
        page.on('dialog', handle_dialog)

        def perform_rolls(num_rolls):
            print(f"\n--- מתחיל סימולציה של {num_rolls} זריקות ---")
            for i in range(num_rolls):
                # Generate random dice values (1-6)
                die1 = random.randint(1, 6)
                die2 = random.randint(1, 6)

                # Fill the input fields
                die1_input.fill(str(die1))
                die2_input.fill(str(die2))

                # Wait for the roll button to become enabled (i.e., not disabled)
                page.wait_for_selector('#rollButton:not([disabled])') 
                
                # Now that the button is enabled, click it
                roll_button.click()
            
            # After all rolls, get the fairness score
            time.sleep(0.5) # Give a small moment for UI to update fairness score
            score = fairness_score_display.text_content()
            print(f"ציון הוגנות לאחר {num_rolls} זריקות: {score}")

            # --- NEW: Get and print the distribution of sums ---
            # Execute JavaScript in the browser context to get the rollCounts array
            # We slice from index 2 because rollCounts[0] and rollCounts[1] are unused (for sums 0 and 1)
            roll_counts_from_browser = page.evaluate("() => window.rollCounts.slice(2)")
            
            print("התפלגות סכומים:")
            for i, count in enumerate(roll_counts_from_browser):
                sum_value = i + 2 # Since slice(2) means index 0 here corresponds to sum 2
                print(f"  סכום {sum_value}: {count} פעמים")
            # --- END NEW ---

        # --- Run Tests ---

        # Test 1: 50 rolls
        perform_rolls(50)
        reset_button.click() # Reset data for the next test

        # Test 2: 100 rolls
        perform_rolls(100)
        reset_button.click() # Reset data for the next test

        # Test 3: 1000 rolls
        perform_rolls(1000)
        # No reset needed after the last test
        
        browser.close() # Close the browser

if __name__ == "__main__":
    try:
        run_dice_fairness_tests()
    except Exception as e:
        print(f"An error occurred during the Playwright test: {e}")
        exit(1) # Exit with a non-zero code to indicate failure

File: diceTest.js
// diceTest.js
const { chromium } = require('playwright');
const path = require('path');

async function runDiceFairnessTests() {
    let browser;
    try {
        browser = await chromium.launch(); // Launch a headless browser
        const page = await browser.newPage();

        // Navigate to the local HTML file
        // Make sure the path is correct relative to where the script runs
        const htmlFilePath = path.resolve(__dirname, 'index.html'); // Assuming your HTML file is named index.html
        await page.goto(`file://${htmlFilePath}`);

        // Selectors for the elements we need to interact with
        const die1Input = page.locator('#die1');
        const die2Input = page.locator('#die2');
        const rollButton = page.locator('#rollButton');
        const resetButton = page.locator('#resetButton');
        const fairnessScoreDisplay = page.locator('#fairnessScoreDisplay');

        // Handle the confirmation dialog for the reset button
        page.on('dialog', async dialog => {
            if (dialog.type() === 'confirm' && dialog.message().includes('לאפס את כל נתוני הזריקות')) {
                await dialog.accept(); // Accept the confirmation
            } else {
                await dialog.dismiss(); // Dismiss other dialogs
            }
        });

        async function performRolls(numRolls) {
            console.log(`\n--- מתחיל סימולציה של ${numRolls} זריקות ---`);
            for (let i = 0; i < numRolls; i++) {
                // Generate random dice values (1-6)
                const die1 = Math.floor(Math.random() * 6) + 1;
                const die2 = Math.floor(Math.random() * 6) + 1;

                // Fill the input fields
                await die1Input.fill(die1.toString());
                await die2Input.fill(die2.toString());

                // Wait for the button to become enabled (due to validation) and then click it
                // FIX: Use locator.waitFor({ state: 'enabled' }) instead of locator.waitForSelector()
                await rollButton.waitFor({ state: 'enabled' }); 
                await rollButton.click();
            }

            // After all rolls, get the fairness score
            // Add a small delay to ensure the UI updates the fairness score
            await page.waitForTimeout(500); 
            const score = await fairnessScoreDisplay.textContent();
            console.log(`ציון הוגנות לאחר ${numRolls} זריקות: ${score}`);
        }

        // --- Run Tests ---

        // Test 1: 50 rolls
        await performRolls(50);
        await resetButton.click(); // Reset data for the next test

        // Test 2: 100 rolls
        await performRolls(100);
        await resetButton.click(); // Reset data for the next test

        // Test 3: 1000 rolls
        await performRolls(1000);
        // No reset needed after the last test
        
    } catch (error) {
        console.error('An error occurred during the Playwright test:', error);
        process.exit(1); // Exit with a non-zero code to indicate failure
    } finally {
        if (browser) {
            await browser.close(); // Close the browser
        }
    }
}

runDiceFairnessTests();
